### PR TITLE
make zookeeper as register and config center

### PR DESCRIPTION
**Pull request by xForMe**
### Ⅰ. Describe what this PR did

merge develop

### CODE REVIEW

File: config/src/main/java/com/alibaba/fescar/config/ZKConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        }
+    }
+
+    @Override
+    public String getTypeName() {
+        return REGISTRY_TYPE;
+    }
+
+    @Override
+    public String getConfig(String dataId, String defaultValue, long timeoutMills) {
```
Comment (slievrly) [inline config/src/main/java/com/alibaba/fescar/config/ZKConfiguration.java]: How to control the timeout?

File: discovery/src/main/java/com/alibaba/fescar/discovery/registry/ZKRegisterServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+            String[] ipAndPort = path.split(IP_PORT_SPLIT_CHAR);
+            if (ipAndPort.length != 2) {
+                throw new IllegalArgumentException("endpoint format should like ip:port");
+            }
+            interSocketAddresses.add(new InetSocketAddress(ipAndPort[0], Integer.valueOf(ipAndPort[1])));
+        });
+        return interSocketAddresses;
+    }
+
+    private ZkClient getClientInstance() {
```
Comment (slievrly) [inline discovery/src/main/java/com/alibaba/fescar/discovery/registry/ZKRegisterServiceImpl.java]: Configuration and registry do not have to be the same type

File: discovery/src/main/java/com/alibaba/fescar/discovery/registry/ZKRegisterServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        return FILE_CONFIG.getConfig(clusterConfigName);
+    }
+
+    private String getServiceGroup(String key){
+        Configuration configuration = ConfigurationFactory.getInstance();
+        String clusterName = PREFIX_SERVICE_ROOT + CONFIG_SPLIT_CHAR + PREFIX_SERVICE_MAPPING + key;
+        return configuration.getConfig(clusterName);
+    }
+
+    private String getIPAndPort(InetSocketAddress address) {
```
Comment (slievrly) [inline discovery/src/main/java/com/alibaba/fescar/discovery/registry/ZKRegisterServiceImpl.java]: should use NetUtils

File: config/src/main/java/com/alibaba/fescar/config/ZKConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    public List<IZkDataListener> getConfigListeners(String dataId) {
+        throw new NotSupportYetException("not support putConfigIfAbsent");
+    }
+
+    public ZkClient getZkClient(){
+        return zkClient;
+    }
+
+
+    private static String getZKAddrFileKey() {
```
Comment (slievrly) [inline config/src/main/java/com/alibaba/fescar/config/ZKConfiguration.java]: this key？

Comment (slievrly) [reply config/src/main/java/com/alibaba/fescar/config/ZKConfiguration.java]: Registry and config configuration are independent of each other

File: discovery/src/main/java/com/alibaba/fescar/discovery/registry/ZKRegisterServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+
+    @Override
+    public void unregister(InetSocketAddress address) throws Exception {
+        String path = ROOT_PATH + getClusterName()+ZK_PATH_SPLIT_CHAR+getIPAndPort(address);
+        getClientInstance().delete(path);
+    }
+
+    @Override
+    public void subscribe(String cluster, IZkChildListener listener) throws Exception {
+        String clusterName = getServiceGroup(cluster);
```
Comment (slievrly) [inline discovery/src/main/java/com/alibaba/fescar/discovery/registry/ZKRegisterServiceImpl.java]: cluster means clusterName

File: discovery/src/main/java/com/alibaba/fescar/discovery/registry/ZKRegisterServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+            return ;
+        }
+        String path = ROOT_PATH  + clusterName;
+        if(!getClientInstance().exists(path)) {return;}
+        getClientInstance().subscribeChildChanges(path,listener);
+    }
+
+    @Override
+    public void unsubscribe(String cluster, IZkChildListener listener) throws Exception {
+        String clusterName = getServiceGroup(cluster);
```
Comment (slievrly) [inline discovery/src/main/java/com/alibaba/fescar/discovery/registry/ZKRegisterServiceImpl.java]: cluster means clusterName

File: config/src/main/java/com/alibaba/fescar/config/ZKConfiguration.java (hunk 4 above)
Comment (xForMe) [reply config/src/main/java/com/alibaba/fescar/config/ZKConfiguration.java]: get zk basic information.such as serverAddr、timeout.

**Comment by jovany-wang** [comment]
What does this PR do? It's better to add title and brief.

**Comment by slievrly** [comment]
Zk will be taken over by @sunrui849

**Pull request merged**